packages/controllers/src/snaps/SnapController.ts

```diff
@@ -268,6 +269,14 @@ export type HandleSnapRpcRequest = {
   handler: SnapController['handleRpcRequest'];
 };
 
+/**
+ * Handles sending an inbound transaction insight message to a snap and returns its result.
+ */
+export type HandleSnapTransactionInsightRequest = {
```

Comment:
Exposing `handleSnapRequest` directly might be better nicer, deprecating onRPC call handler

Reply:
Hmm, yeah I agree. We'll have to just keep adding here in the future as we add more handlers anyway, best to generalize and then we can just pass in the handler type through the front-end.